src/lib/data-api.ts

```diff
@@ -70,7 +70,7 @@ export class DataApi {
       if (file) {
         content = await file.read();
         content = interpolateTemplate(content, {
-          title: () => (record.values["name"] as string | undefined) ?? "",
+          title: () => getNameFromPath(record.id),
```

Comment:
Record id is formatted by the function `createDataRecord()` in file `data-api.ts`
```typescript
  return {
    id: path + "/" + name + ".md",
    values: values ?? {},
  };
```
Function `getNameFromPath()` only provides simple string operations. An inline version is also considerable.